tests/manipulator_test.py

```diff
@@ -0,0 +1,321 @@
+import unittest
+import opentuner
+import random
+import numpy
+from opentuner.search import manipulator
+
+saved_random = random.random
+saved_randint = random.randint
+saved_randuniform = random.uniform
+
+def restore_random():
+    random.random = saved_random
+
+def restore_randint():
+    random.randint = saved_randint
+
+def restore_randuniform():
+    random.uniform = saved_randuniform
+
+# helper method for overriding manipulator.random
+# nums is a list of numbers
+def override_random(nums):
+    f = fake_random(nums)
+    random.random = lambda : f.next()
```

Comment:
A better way to do this would be with mocking.  Don't monkey patch standard libraries.
http://www.toptal.com/python/an-introduction-to-mocking-in-python